catroid/src/androidTest/java/org/catrobat/catroid/test/utils/CrashReporterTest.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2017 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.test.utils;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.preference.PreferenceManager;
+import android.support.test.InstrumentationRegistry;
+import android.test.AndroidTestCase;
+
+import org.catrobat.catroid.ui.SettingsActivity;
+import org.catrobat.catroid.utils.CrashReporter;
+import org.junit.After;
+import org.junit.Before;
+
+public class CrashReporterTest extends AndroidTestCase {
+
+	private Context context;
+	private SharedPreferences sharedPreferences;
+	private SharedPreferences.Editor editor;
+	private Exception exception;
+
+	@Before
+	public void setUp() {
+		context = InstrumentationRegistry.getTargetContext();
+		sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
+		editor = sharedPreferences.edit();
+		editor.clear();
+		editor.putBoolean(SettingsActivity.SETTINGS_CRASH_REPORTS, true);
+		editor.commit();
+		exception = new RuntimeException("Error");
+		CrashReporter.setIsCrashReportEnabled(true);
+		CrashReporter.initialize(context);
+	}
+
+	@After
+	public void tearDown() {
+		CrashReporter.setIsCrashReportEnabled(false);
+	}
+
+	public void testCrashlyticsUninitializedOnAnonymousReportDisabled() {
+		editor.putBoolean(SettingsActivity.SETTINGS_CRASH_REPORTS, false);
+		editor.commit();
+
+		assertFalse(CrashReporter.initialize(context));
+	}
+
+	public void testCrashlyticsInitializedOnAnonymousReportEnabled() {
+		assertTrue(CrashReporter.initialize(context));
+	}
+
+	public void testCrashlyticsUninitializedOnCrashReportDisabled() {
+		CrashReporter.setIsCrashReportEnabled(false);
+
+		assertFalse(CrashReporter.initialize(context));
+	}
+
+	public void testUnhandledExceptionStoredOnCrashReportEnabled() {
+		CrashReporter.storeUnhandledException(exception);
+
+		assertTrue(!sharedPreferences.getString(CrashReporter.EXCEPTION_FOR_REPORT, "").isEmpty());
```

Comment:
This could be `assertFalse(sharedPreferences.getString(CrashReporter.EXCEPTION_FOR_REPORT, "").isEmpty());` for better readability.

This is valid for all further occurrences.